python/fusion_engine_client/messages/ros.py

```diff
@@ -0,0 +1,352 @@
+
+    _FORMAT = '< 3d 4d'
+    _SIZE: int = struct.calcsize(_FORMAT)
+
+    def __init__(self):
+        # The time of the message, in P1 time (beginning at power-on)
+        self.p1_time = Timestamp()
+        # relative ENU position (see ros.h messages)
+        self.position_rel_m = np.full((3,), np.nan)
+        # The platform orientation, represented as a quaternion with the scalar
+        # component last (x, y, z, w)
+        self.orientation = np.full((4,), np.nan)
+
+    def get_type(self) -> MessageType:
+        return PoseMessage.MESSAGE_TYPE
+
+    def pack(self, buffer: bytes = None, offset: int = 0, return_buffer: bool = True) -> (bytes, int):
+        if buffer is None:
+            buffer = bytes(self.calcsize())
+
+        initial_offset = offset
+
+        offset += self.p1_time.pack(buffer, offset, return_buffer=False)
+
+        struct.pack_into(PoseMessage._FORMAT, buffer, offset,
+                         self.position_rel_m[0], self.position_rel_m[1], self.position_rel_m[2],
+                         self.orientation[0], self.orientation[1], self.orientation[2], self.orientation[3])
+
+        if return_buffer:
+            return buffer
+        else:
+            return offset - initial_offset
+
+    def unpack(self, buffer: bytes, offset: int = 0) -> int:
+        initial_offset = offset
+
+        offset += self.p1_time.unpack(buffer, offset)
+        (self.position_rel_m[0], self.position_rel_m[1], self.position_rel_m[2],
+         self.orientation[0], self.orientation[1], self.orientation[2], self.orientation[3]) = \
+            struct.unpack_from(PoseMessage._FORMAT,
+                               buffer=buffer, offset=offset)
+        offset += PoseMessage._SIZE
+        return offset - initial_offset
+
+    def __repr__(self):
+        return '%s @ %s' % (self.MESSAGE_TYPE.name, self.p1_time)
+
+    def __str__(self):
+        string = 'ROS Pose message @ P1 time %s\n' % str(self.p1_time)
+        string += '  Position : %.6f, %.6f, %.6f \n' % tuple(self.position_rel_m)
+        string += '  Orientation: %.12f, %.12f, %.12f, %.12f \n' % tuple(self.orientation)
+        return string
+
+    @classmethod
+    def calcsize(cls) -> int:
+        return Timestamp.calcsize() + PoseMessage._SIZE
+
+    @classmethod
+    def to_numpy(cls, messages):
+        result = {
+            'p1_time': np.array([float(m.p1_time) for m in messages]),
+            'position_rel_m': np.array([m.position_rel_m for m in messages]).T,
+            'orientation': np.array([m.orientation for m in messages]).T,
+        }
+        return result
+
+
+class GPSFixMessage(MessagePayload):
+    """!
+    @brief ROS `GPSFix` message (MessageType::ROS_GPS_FIX).
+    """
+    MESSAGE_TYPE = MessageType.ROS_GPS_FIX
+
+    _FORMAT = '< ddd ddd ddd d ddddd dddddddddd 9d B 3B'
+    _SIZE: int = struct.calcsize(_FORMAT)
+
+    def __init__(self):
+        self.p1_time = Timestamp()
+        # WGS-84 Geodetic Position
```

Comment:
These are not necessarily WGS84

Reply:
Yeah, the ROS message definition doesn't actually specify. I'll fix this comment.